File: src/main/java/com/sothawo/springdataelastictest/foo/FooHandler.java
/*
 * (c) Copyright 2023 sothawo
 */
package com.sothawo.springdataelastictest.foo;

import org.springframework.stereotype.Component;
import org.springframework.web.reactive.function.server.ServerRequest;
import org.springframework.web.reactive.function.server.ServerResponse;
import reactor.core.publisher.Mono;

import static org.springframework.web.reactive.function.server.ServerResponse.*;

/**
 * @author P.J. Meisch (pj.meisch@sothawo.com)
 */
@Component
public class FooHandler {
	private FooService service;

	public FooHandler(FooService service) {
		this.service = service;
	}

	public Mono<ServerResponse> test(ServerRequest serverRequest) {
		return ok().body(service.test3(), Foo.class);
	}
}

File: src/main/kotlin/com/sothawo/springdataelastictest/so/FooHandler.kt
package com.sothawo.springdataelastictest.so

import org.springframework.stereotype.Component
import org.springframework.web.reactive.function.server.ServerRequest
import org.springframework.web.reactive.function.server.ServerResponse
import org.springframework.web.reactive.function.server.bodyValueAndAwait
import org.springframework.web.reactive.function.server.json

/**
 * @author P.J. Meisch (pj.meisch@sothawo.com)
 */
@Component
class FooHandler(private val service: FooService) {
	suspend fun test(request: ServerRequest): ServerResponse =
		ServerResponse.ok().json().bodyValueAndAwait(service.test())
}
